### PR TITLE
Add permissions integration tests

### PR DESCRIPTION
**Pull request by dhovart**
I've added some integration tests to help with the refactoring and make sure I'm not pushing  regressions.
I've corrected some cases were users could add annotations or collaborators wihthout have the right to.

It  would be great if you could review the different cases I've defined and tell me if they're ok.

### CODE REVIEW

File: controller/checkAccess/checkAccess.js [hunk 1]
```diff
@@ -7,6 +7,9 @@
 /* eslint-disable valid-jsdoc */
 /* eslint-disable max-depth */
 /* eslint-disable max-lines */
+
+const _ = require('lodash');
```
Comment (r03ert0) [inline controller/checkAccess/checkAccess.js:11]: we've never used lodash before, but I'm happy to start. What are the main things you use it for?

Comment (dhovart) [reply controller/checkAccess/checkAccess.js:11]: Ah, sorry, I didn't see your message.
I think it's a really handy library. There's a lot of helpers that make working with collections especially great, be they arrays or objects. It's to be used in a functional way, which I think is nice as it's more declarative / makes your intent more clear.

You may read posts about people saying that it's now unnecessary and adds bloat but it comes mostly from frontend developers concerned with the size of their final bundle working in an E6 context. For node and js < ES6 I think it makes sense to use it.

**Comment by dhovart** [comment]
Tests should not fail anymore after https://github.com/neuroanatomy/NeuroWebLab/pull/4 is merged

**Comment by katjaq** [comment]
https://github.com/neuroanatomy/NeuroWebLab/pull/4 has been merged now

**Comment by katjaq** [comment]
Awesome! 🚀 🚀 🚀

**Pull request merged**